From 1eabce3a9c15c5caf022e71c1959e8c8f4819fad Mon Sep 17 00:00:00 2001
From: "Anthony G. Basile" <blueness@gentoo.org>
Date: Thu, 3 Mar 2016 20:48:43 -0500
Subject: [PATCH] targets/support/chroot-functions.sh: correct quotes to allow
 expansion of $1

This is a backport of commit b2473eff.  The single quotes prevented the proper
expansion of $1 in setup_pkgmgr().  Since this is usually called as
`setup_pkgmgr "build"` during stage1 builds, it is surprising this was
not caught earlier.
---
 targets/support/chroot-functions.sh | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/targets/support/chroot-functions.sh b/targets/support/chroot-functions.sh
index 3b7f77b..6c1a31c 100755
--- a/targets/support/chroot-functions.sh
+++ b/targets/support/chroot-functions.sh
@@ -172,12 +172,12 @@ setup_pkgmgr(){
 	# Use --update or portage might just waste time/cycles and reinstall the same version.
 	# Use --newuse to make sure it rebuilds with any changed use flags.
 	if [ -n "$1" ];then
-		echo "Adding USE='${USE} $1' to make.conf for portage build"
-		[ -e /etc/portage/make.conf ] && echo 'USE="${USE} $1"' >> /etc/portage/make.conf
+		echo "Adding USE=\"${USE} $1\" to make.conf for portage build"
+		[ -e /etc/portage/make.conf ] && echo "USE=\"\${USE} $1\"" >> /etc/portage/make.conf
 		run_merge --oneshot --update --newuse sys-apps/portage
-		sed -i '/USE="${USE} $1"/d' /etc/portage/make.conf
+		sed -i "/USE=\"\${USE} $1\"/d" /etc/portage/make.conf
 	else
-		echo "Updating portage with USE='${USE}'"
+		echo "Updating portage with USE=\"${USE}\""
 		run_merge --oneshot --update --newuse sys-apps/portage
 	fi
 }
-- 
2.4.10

